### ISSUE TITLE
Failure while translating Erlang's logger event 

### ISSUE THREAD

**Issue by ConnorRigby** [elixir-lang/elixir#8126]
I hit an error using the AMQP Elixir wrapper around the Erlang `:amqp_client`:

```
15:24:26.514 [error] Failure while translating Erlang's logger event
** (FunctionClauseError) no function clause matching in Logger.Translator.child_info/2
    (logger) lib/logger/translator.ex:320: Logger.Translator.child_info(:debug, [name: :connection, mfargs: {:amqp_gen_connection, :start_link, [#PID<0.18903.2>, {:amqp_params_network, "device_6", "SUPER SECRET TOKEN", "vbzcxsqr", 'brisk-bear.rmq.cloudamqp.com', 5672, 0, 0, 10, 60000, :none, [&:amqp_auth_mechanisms.plain/3, &:amqp_auth_mechanisms.amqplain/3], [], []}]}, restart_type: :intrinsic, shutdown: :brutal_kill, child_type: :worker])
    (logger) lib/logger/translator.ex:294: Logger.Translator.report_supervisor/2
    (logger) lib/logger/erlang_handler.ex:104: Logger.ErlangHandler.translate/6
    (logger) lib/logger/erlang_handler.ex:97: Logger.ErlangHandler.translate/5
    (logger) lib/logger/erlang_handler.ex:27: anonymous fn/3 in Logger.ErlangHandler.log/2
    (logger) lib/logger.ex:764: Logger.normalize_message/2
    (logger) lib/logger.ex:625: Logger.bare_log/3
    (kernel) logger_backend.erl:51: :logger_backend.call_handlers/3
```

I had originally opened this issue [here](https://github.com/pma/amqp/issues/107) but after actually looking at it it looks like it is either something missing in the [Elixir Logger translator code](https://github.com/elixir-lang/elixir/blob/ea76fe8224e3614bdc14bec17c31568b6a4bb576/lib/logger/lib/logger/translator.ex#L328-L344)

Feel free to close if you believe this is an issue with one of the underlying libs.

**Comment by josevalim** [comment]
@ConnorRigby what is your Elixir and Erlang/OTP versions? It is very important to follow the issues template, because this may be a bug that has already been fixed. Thanks!

**Comment by ConnorRigby** [comment]
Ah man very sorry. I had it in my clipboard, but forgot to paste it:

* erlang otp 21.0.4
* elixir 1.6.6-otp-21

**Comment by josevalim** [comment]
Thanks! :+1: It is fixed on the released v1.7.0+ and a fix is also on the v1.6 branch! If you see the code, you can see that we have added a pokemon clause.
